src/analysis/mean_inequalities.lean

```diff
@@ -125,6 +125,48 @@ begin
       { rw [exp_log hz] } } }
 end
 
+theorem geom_mean_eq_arith_mean_weighted_of_constant (w z : ι → ℝ) (x : ℝ) (hw : ∀ i ∈ s, 0 ≤ w i)
+  (hw' : ∑ i in s, w i = 1) (hz : ∀ i ∈ s, 0 ≤ z i) (hx : ∀ i ∈ s, w i ≠ 0 → z i = x) :
```

Comment:
Phrased this way, both sides are equal to `x`. And it might be useful to prove that as two separate lemmas first.
I can also imagine having a version that doesn't mention `x`, but requires `∀ i ∈ s, ∀ j ∈ s, w i ≠ 0 → w j ≠ 0 → z i = z j` instead.